nearprotocol/nearcore
# fix(epoch manager): Fix fishermen unstake

Currently if fishermen unstake by staking 0 it causes inconsistency in the epoch manager: while the stake will be returned normally, unstaked fishermen remain fishermen in epoch manager. This PR fixes this issue.

Test plan
---------
* Add `test_fishermen_unstake` in epoch manager to make sure the field `fishermen` is updated correctly.
* Add `test_fishermen_unstake` in near runtime to make sure that fishermen stake is properly returned when they unstake.
